perceval/backends/askbot.py

```diff
@@ -0,0 +1,534 @@
+    """
+    version = '0.1.0'
+
+    def __init__(self, url, tag=None):
+        origin = url
+
+        super().__init__(origin, tag=tag)
+        self.url = url
+        self.client = AskbotClient(url)
+        self.ab_parser = AskbotParser()
+
+    @metadata
+    def fetch(self, from_date=DEFAULT_DATETIME):
+        """Fetch the questions/answers from the repository.
+
+        The method retrieves, from an Askbot site, the questions and answers
+        updated since the given date.
+
+        :param from_date: obtain questions/answers updated since this date
+
+        :returns: a generator of items
+        """
+
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        from_date = from_date.timestamp()
+
+        npages = 1
+        next_request = True
+
+        while next_request:
+            whole_page = self.client.get_api_questions(npages)
+            raw_questions = json.loads(whole_page)
+            questions = raw_questions['questions']
+            tpages = raw_questions['pages']
+
+            logger.debug("Fetching questions from '%s': page %s/%s",
+                         self.url, npages, tpages)
+
+            for question in questions:
+                updated_at = int(question['last_activity_at'])
+                if updated_at > from_date:
+                    bs_question = self.fetch_html_question(question)
+                    logger.debug("Fetching HTML question %s", question['id'])
+                    question = self.ab_parser.parse_question(question, bs_question)
+                    yield question
+
+            if npages == tpages:
+                next_request = False
+
+            npages = npages + 1
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an Askbot question item."""
+
+        return str(item['id'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from an Askbot item.
+
+        The timestamp is extracted from 'last_activity_at' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return float(item['last_activity_at'])
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from an Askbot item.
+
+        This backend only generates one type of item which is
+        'question'.
+        """
+        return 'question'
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return True
+
+    @classmethod
+    def has_caching(cls):
+        """Returns whether it supports caching items on the fetch process.
+
+        :returns: this backend supports items cache
+        """
+        return False
+
+    def fetch_html_question(self, question):
```

Comment:
This should be a private method and I think it should be renamed to `__fetch_question`.